internal/services/firewall/firewall_policy_resource.go

```diff
@@ -374,6 +548,69 @@ func expandFirewallPolicyDNSSetting(input []interface{}) *network.DNSSettings {
 	return output
 }
 
+func expandFirewallPolicyIntrusionDetection(input []interface{}) *network.FirewallPolicyIntrusionDetection {
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+
+	raw := input[0].(map[string]interface{})
+
+	signatureOverrides := []network.FirewallPolicyIntrusionDetectionSignatureSpecification{}
+	for i, v := range signatureOverrides {
+		signatureOverrides[i].ID = v.ID
+		signatureOverrides[i].Mode = v.Mode
+	}
+
+	trafficBypass := []network.FirewallPolicyIntrusionDetectionBypassTrafficSpecifications{}
+
+	for i, v := range trafficBypass {
+		trafficBypass[i].Name = v.Name
+		trafficBypass[i].Description = v.Description
+		trafficBypass[i].Protocol = v.Protocol
+		trafficBypass[i].SourceAddresses = v.SourceAddresses
+		trafficBypass[i].DestinationAddresses = v.DestinationAddresses
+		trafficBypass[i].DestinationPorts = v.DestinationPorts
+		trafficBypass[i].SourceIPGroups = v.SourceIPGroups
+		trafficBypass[i].DestinationIPGroups = v.DestinationIPGroups
+	}
+
+	return &network.FirewallPolicyIntrusionDetection{
+		Mode: network.FirewallPolicyIntrusionDetectionStateType(raw["mode"].(string)),
+		Configuration: &network.FirewallPolicyIntrusionDetectionConfiguration{
+			SignatureOverrides:    &signatureOverrides,
+			BypassTrafficSettings: &trafficBypass,
+		},
+	}
+
+}
+
+func expandFirewallPolicyTransportSecurity(input []interface{}) *network.FirewallPolicyTransportSecurity {
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+
+	raw := input[0].(map[string]interface{})
+
+	return &network.FirewallPolicyTransportSecurity{
+		CertificateAuthority: &network.FirewallPolicyCertificateAuthority{
+			KeyVaultSecretID: utils.String(raw["key_vault_secret_id"].(string)),
+			Name:             utils.String(raw["name"].(string)),
+		},
+	}
+}
+
+func expandFirewallPolicyIdentity(input []interface{}) *network.ManagedServiceIdentity {
+	if len(input) == 0 {
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
+
+	return &network.ManagedServiceIdentity{
+		Type: network.ResourceIdentityType(v["type"].(string)),
```

Comment:
It looks like the `principal_id`, `tenant_id` and `identity_ids` properties need to be added here?

Reply:
thanks for all your suggestions, this should be addressed in commit 62a615471cf92cce55940d59ce37122dce501bfe.